ExampleFhirClient/Program.cs

```diff
@@ -0,0 +1,377 @@
+			{
+				Console.WriteLine( "response status = " + bundleResponseEntry.Response.Status );
+				Console.WriteLine( "location = " + bundleResponseEntry.Response.Location );
+
+				var resource = fhirClient.Get( bundleResponseEntry.Response.Location );
+				if ( resource is FhirModel2.Patient )
+				{
+					patient = (FhirModel2.Patient) resource;
+				} else if ( resource is FhirModel2.DiagnosticReport )
+				{
+					report = (FhirModel2.DiagnosticReport) resource;
+				}
+			}
+
+			return patient;
+		}
+
+		private static FhirModel2.Patient CreatePatient( string patientId )
+		{
+			return new FhirModel2.Patient
+			{
+				Identifier = new List<FhirModel2.Identifier> {
+					new FhirModel2.Identifier
+					{
+						Value = patientId
+					}
+				},
+				Name = new List<FhirModel2.HumanName> {
+					new FhirModel2.HumanName
+					{
+						Family =  new[] { "TestFhirPost" },
+						Given = new[] { "Miss " + patientId.ToString() },
+					}
+				},
+				Gender = FhirModel.AdministrativeGender.Female,
+			};
+		}
+
+		private static FhirModel2.DiagnosticReport CreateReport( DateTime reportDate )
+		{
+			return new FhirModel2.DiagnosticReport
+			{
+				Identifier = new List<FhirModel2.Identifier>
+				{
+					new FhirModel2.Identifier( "http://mysystem.org", "PostViaFhir " + Guid.NewGuid() )
+				},
+
+				Effective = new FhirModel.FhirDateTime( reportDate ),
+				Issued = reportDate,
+				Result = new List<FhirModel2.ResourceReference>(),
+				Category = new FhirModel.CodeableConcept( "http://terminology.hl7.org/CodeSystem/v2-0074", "MB", "MB Display" ),
+				PresentedForm = new List<FhirModel.Attachment>
+				{
+					new FhirModel.Attachment
+					{
+						ContentType = "text/plain; charset=UTF-8",
+						Data = Encoding.UTF8.GetBytes("More Lab Report Content")
+					}
+				},
+				Code = new FhirModel.CodeableConcept( "http://loinc.org", "632-0" ),
+				Status = FhirModel2.DiagnosticReportStatus.Final
+			};
+		}
+
+		public FhirModel2.Bundle CreateTransactionBundle(
+			FhirModel.Resource resource,
+			string requestUrl,
+			FhirModel.HTTPVerb requestMethod
+		)
+		{
+			var bundle = new FhirModel2.Bundle { Type = FhirModel.BundleType.Transaction };
+			bundle.Entry.Add( CreateTransactionBundleEntry( resource, requestUrl, requestMethod ) );
+			return bundle;
+		}
+
+		public static FhirModel2.Bundle.EntryComponent CreateTransactionBundleEntry(
+			FhirModel.Resource resource,
+			string requestUrl,
+			FhirModel.HTTPVerb requestMethod
+		)
+		{
+			return new FhirModel2.Bundle.EntryComponent
+			{
+				Request = new FhirModel2.Bundle.RequestComponent { Url = requestUrl, Method = requestMethod },
+				FullUrl = "urn:uuid:" + Guid.NewGuid(),
+				Resource = resource
+			};
+		}
+
+		// post via http client using resource in json string.  return data is limited, does return error info, but not newly created ID's. 
+		private static void PostJsonReport( string accessToken, string fhirEndPoint, string patientId )
```

Comment:
This is different from all the other examples - does not use FhirClient - is that intentional?

Reply:
this example is different and it was intentional.  I encounterd users at a site that wanted to post via json.  Thought it would be helpful to see both FhirClient and Json approach.